src/main/ruby/core/truffle/ffi/ffi_struct.rb

```diff
@@ -237,6 +237,11 @@ def initialize(pointer=nil, *spec)
 
       if pointer
         @pointer = pointer
+      elsif block_given?
+        MemoryPointer.new size do |pointer|
+          @pointer = pointer
```

Comment:
I'm a bit confused on how this is free-by-scope. Isn't the pointer still captured? Is it just invalid after the block exits?

Reply:
Yes that's the existing `MemoryPointer` API - if you pass block to `new` it's freed after the block returns.

Reply:
Fixed.

Reply:
Should you clear `@pointer` after the block exits though? It seems to me it could be confusing to have a non-nil `@pointer` that references a pointer without any allocated memory.

Reply:
It's no different to how FFI pointers work already. It's exactly the same interface as `MemoryPointer`, just applied to structs.